File: src/main/java/com/github/spb/tget/demo/repository/dbRepository/CountryDbRepository.java
package com.github.spb.tget.demo.repository.dbRepository;

import com.github.spb.tget.demo.repository.DbRepository;

public class CountryDbRepository extends DbRepository {

    private CountryDbRepository() {
        this.entityName = "Country";
    }

    public static CountryDbRepository create() {
        return new CountryDbRepository();
    }
}

File: Service/src/main/java/com/github/spb/tget/demo/repository/dbRepository/CountryDbRepository.kt
package com.github.spb.tget.demo.repository.dbRepository

import com.github.spb.tget.demo.data.CountryEntity

class CountryDbRepository private constructor() : DbRepository<CountryEntity>() {

    init {
        this.entityName = "CountryEntity"
    }

    companion object {

        fun create(): CountryDbRepository {
            return CountryDbRepository()
        }
    }
}
